sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/InputFilePredicate.java

```diff
@@ -21,20 +21,7 @@
 
 import org.sonar.api.batch.fs.InputFile;
 
-/**
- * @since 4.2
- */
-class StatusPredicate extends AbstractFilePredicate {
-
-  private final InputFile.Status status;
-
-  StatusPredicate(InputFile.Status status) {
-    this.status = status;
-  }
-
-  @Override
-  public boolean apply(InputFile f) {
-    return status == f.status();
-  }
-
+@FunctionalInterface
+public interface InputFilePredicate {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Drop this interface in favor of "java.util.function.Predicate<InputFile>". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1711)